lib/wheel.c

```diff
@@ -149,6 +155,73 @@ int wheel_add_item(struct timer_wheel *wheel, void *item)
 			   slot % wheel->slots);
 	listnode_add(wheel->wheel_slot_lists[slot % wheel->slots], item);
 
+	/*
+	 * Figure out how many slots we should skip with the newly added
+	 * item
+	 */
+	while (list_isempty(
```

Comment:
don't need to loop recomputing `slots_to_skip` - you know that nothing could have been inserted between current and target because there's only one "insert" api? and we don't try to recompute when an item is deleted.